bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/internal/server/CLanguageServerStreamConnectionProvider.java

```diff
@@ -65,8 +65,8 @@ protected ProcessBuilder createProcessBuilder() {
 	public void start() throws IOException {
 		super.start();
 		if (logEnabled() && getLogProvider().isPresent()) {
-			errorStreamPipeStopper = new AsyncStreamPipe().pipeTo(new BufferedInputStream(getErrorStream()),
-					getLogProvider().get().getOutputStream());
+			errorStreamPipeStopper = new AsyncStreamPipe().pipeTo("LS stderr pipe", //$NON-NLS-1$
```

Comment:
I like that you added a thread name. Can you mention CDT in it?

```suggestion
			errorStreamPipeStopper = new AsyncStreamPipe().pipeTo("CDT LS stderr pipe", //$NON-NLS-1$
```